### PR TITLE
chore: add typescript

### PR DESCRIPTION
**Pull request by pavandv**
added Typescript Config and changed babel configuration to transpile TS, JS files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -151,7 +159,7 @@
     "start:ui": "node lib/bin/styleguidist.js server",
     "lint": "eslint . --cache --fix",
     "flow": "flow",
-    "compile": "babel --delete-dir-on-start --ignore '**/*.spec.js','**/__*__/*' -d lib/ src/",
+    "compile": "babel --delete-dir-on-start --ignore '**/*.spec.js','**/__*__/*' --extensions '.js','.ts' -d lib/ src/",
```
Comment (sapegin) [inline package.json]: We need .tsx as well.

```suggestion
    "compile": "babel --delete-dir-on-start --ignore '**/*.spec.js','**/__*__/*' --extensions '.js,.ts,.tsx' -d lib/ src/",
```

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "es2015",
+    "jsx": "react",
```
Comment (smashercosmo) [inline tsconfig.json]: Why not 'preserve'? JSX compilation is handled by babel.

File: package.json (hunk 1 above)
Comment (pavandv) [reply package.json]: Done 👍

File: tsconfig.json (hunk 2 above)
Comment (pavandv) [reply tsconfig.json]: Preserve converts `.ts/.tsx` files to `.jsx`. And as we are using tsconfig only for type-checking, 'preserve/react' doesn't matter.

Comment (sapegin) [reply tsconfig.json]: What's the default value? Can we just drop it?

Comment (pavandv) [reply tsconfig.json]: Default is `preserve`. If we drop it, all files should be renamed to `.tsx` instead of `.ts`.

**Pull request merged**